packages/app-root/src/app/groups/[groupId]/GroupStatsContainer.js

```diff
@@ -1,16 +1,77 @@
 'use client'
 
 import { GroupStats } from '@zooniverse/user'
+import { useRouter } from 'next/navigation'
 import { useContext } from 'react'
 
 import { PanoptesAuthContext } from '../../../contexts'
 
 function GroupStatsContainer({
+  endDate,
   groupId,
-  joinToken
+  joinToken,
+  projectId,
+  startDate
 }) {
   const { adminMode, isLoading, user } = useContext(PanoptesAuthContext)
 
+  const router = useRouter()
+  
+  // set end date per query params or default to today
+  let selectedEndDate = endDate
+  if (!selectedEndDate) {
+    selectedEndDate = new Date().toISOString().substring(0, 10)
+  }
+  // set start date per query params or default to 7 days ago
+  let selectedStartDate = startDate
+  if (!selectedStartDate) {
+    selectedStartDate = new Date(new Date().setDate(new Date().getDate() - 6)).toISOString().substring(0, 10)
```

Comment:
This might be more readable, and faster since it only uses one new Date object.
```suggestion
    const today = newDate()
    const 7DaysAgo = today.setUTCDate(today.getUTCDate() - 7)
    selectedStartDate = 7DaysAgo.toISOString().substring(0, 10)
```

Reply:
Yes! I forgot to include the getUTCDate here. I think I've included it in other spots (like BarChart's [`getCompleteData`](packages/lib-user/src/components/shared/BarChart/helpers/getCompleteData.js)), but please let me know if you notice any other places I've missed it. Thank you!

Reply:
👍🏻

By the way, `new Date` is redundant here, since `date.setUTCDate()` returns a Date object. 

https://github.com/zooniverse/front-end-monorepo/blob/db295fdd5a076c12f47911fe37401563c6d4f32c/packages/lib-user/src/components/shared/BarChart/helpers/getCompleteData.js#L13

Reply:
Just realised my code suggestion was wrong, so I rewrote it. I always assume, wrongly, that Date methods return the modified Date.

Reply:
Just added [1145d22](https://github.com/zooniverse/front-end-monorepo/pull/6160/commits/1145d2244dcab2c28b9495d5c9e666b67b44ae3f) and [9b55f9f](https://github.com/zooniverse/front-end-monorepo/pull/6160/commits/9b55f9fa94347025ac7a6c1818bf0574b762413e) to refactor as noted as well as a few other places.